src/library/scala/collection/Iterable.scala

```diff
@@ -696,10 +694,10 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     *  @return        a new $coll containing pairs consisting of corresponding elements of this $coll and `that`.
     *                 The length of the returned collection is the minimum of the lengths of this $coll and `that`.
     */
-  def zip[B](that: Iterable[B]): CC[(A @uncheckedVariance, B)] = fromIterable(new View.Zip(this, that))
+  def zip[B](that: Iterable[B]): CC[(A @uncheckedVariance, B)] = iterableFactory.from(new View.Zip(this, that))
```

Comment:
We could probably take an `IterableOnce` here too (although this was not the case in 2.12).